include/openthread/tcp.h

```diff
@@ -0,0 +1,589 @@
+ */
+otError otTcpActiveSendByExtension(otInstance *aInstance, otTcpActiveSocket *aSocket, size_t aNumBytes, uint32_t aFlags);
+
+/**
+ * Provides the application with a linked buffer chain referencing data
+ * currently in the TCP receive buffer. The linked buffer chain is valid until
+ * the "receive ready" callback is next invoked, or until the next call to
+ * @p otTcpActiveReceiveContiguify or @p otTcpActiveCommitReceive.
+ *
+ * @param[in]   aInstance  A pointer to the OpenThread instance to which the socket belongs.
+ * @param[in]   aSocket    A pointer to the active TCP socket structure representing the TCP endpoint on which to receive data.
+ * @param[out]  aBuffer    A pointer to the linked buffer chain referencing data currently in the receive buffer.
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the operation.
+ */
+otError otTcpActiveReceiveByReference(otInstance *aInstance, otTcpActiveSocket *aSocket, const otLinkedBuffer **aBuffer);
+
+/**
+ * Reorganizes the receive buffer to be entirely contiguous in memory
+ *
+ * This is optional; an application can simply traverse the linked buffer
+ * chain obtained by calling @p otTcpActiveReceiveByReference. Some
+ * applications may wish to call this function to make the receive buffer
+ * contiguous to simplify their data processing, but this comes at the expense
+ * of CPU time to reorganize the data in the receive buffer.
+ *
+ * @param[in]   aInstance  A pointer to the OpenThread instance to which the socket belongs.
+ * @param[in]   aSocket    A pointer to the active TCP socket structure representing the TCP endpoint on which to receive data.
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the operation.
+ */
+otError otTcpActiveReceiveContiguify(otInstance *aInstance, otTcpActiveSocket *aSocket);
+
+/**
+ * Informs the TCP stack that the application has finished processing
+ * @p aNumBytes bytes of data at the start of the receive buffer and that the
+ * TCP stack need not continue maintaining those bytes in the receive buffer.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the socket belongs.
+ * @param[in]  aSocket    A pointer to the active TCP socket structure representing the TCP endpoint on which to receive data.
+ * @param[in]  aNumBytes  The number of bytes consumed.
+ * @param[in]  aFlags     Flags specifying options for this operation (none yet).
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the receive operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the receive operation.
+ */
+otError otTcpActiveCommitReceive(otInstance *aInstance, otTcpActiveSocket *aSocket, size_t aNumBytes, uint32_t aFlags);
+
+/**
+ * This enumeration defines which sides of the connection to shut down in a
+ * call to @p otTcpActiveShutdown.
+ */
+typedef enum otTcpActiveShutdownHow {
```

Comment:
```suggestion
typedef enum otTcpActiveShutdownMode {
```

Reply:
I'm OK with this.

Reply:
I'll change this in my own commit so i can also update the code that uses this structure name.

Reply:
OK, I pushed an update to address this.